src/client/Microsoft.Identity.Client/Http/HttpManager.cs

```diff
@@ -28,6 +28,7 @@ internal class HttpManager : IHttpManager
         protected readonly IMsalHttpClientFactory _httpClientFactory;
         private readonly Func<HttpResponse, bool> _retryCondition;
         public long LastRequestDurationInMs { get; private set; }
+        public HttpClientHandler HttpClientHandler { get; set; }
```

Comment:
It's going to be tough to reason about this class if it becomes mutable. Can you make it immutable by moving this to a ctor?